src/main/java/org/opensearch/knn/index/mapper/VectorTransformer.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.mapper;
+
+/**
+ * Defines operations for transforming vectors in the k-NN search context.
+ * Implementations can modify vectors while preserving their dimensional properties
+ * for specific use cases such as normalization, scaling, or other transformations.
+ *
+ * <p>This interface provides default implementations that pass through the original
+ * vector without modification. Implementing classes should override these methods
+ * to provide specific transformation logic.
+ */
+public interface VectorTransformer {
+
+    /**
+     * Transforms a float vector into a new vector of the same type.
```

Comment:
Will this be in place or always a copy?

Reply:
IMO, Transformer should generally return a new value rather than modifying the original value it receives, to preserve immutability.